library/ssl_tls.c

```diff
@@ -4551,6 +4568,12 @@ int mbedtls_ssl_setup( mbedtls_ssl_context *ssl,
     ssl->out_msg = NULL;
 #endif /* MBEDTLS_SSL_USE_MPS */
 
+#if defined(MBEDTLS_ZERO_RTT) && defined(MBEDTLS_SSL_SRV_C)
+    if( conf->endpoint == MBEDTLS_SSL_IS_SERVER )
+    {
+        mbedtls_free( ssl->early_data_server_buf );
```

Comment:
We have to reliably zeroize this buffer to avoid a data leak. Use `mbedtls_platform_zeroize()` prior to the call to `free()`.